api/internal/managers/app/install/install.go

```diff
@@ -0,0 +1,69 @@
+package install
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/replicatedhq/embedded-cluster/api/internal/utils"
+	kotscli "github.com/replicatedhq/embedded-cluster/cmd/installer/kotscli"
+	"github.com/replicatedhq/embedded-cluster/pkg-new/constants"
+	"github.com/replicatedhq/embedded-cluster/pkg/netutils"
+	kotsv1beta1 "github.com/replicatedhq/kotskinds/apis/kots/v1beta1"
+	kyaml "sigs.k8s.io/yaml"
+)
+
+// Install installs the app with the provided config values
+func (m *appInstallManager) Install(ctx context.Context, configValues kotsv1beta1.ConfigValues) error {
+	license := &kotsv1beta1.License{}
+	if err := kyaml.Unmarshal(m.license, license); err != nil {
+		return fmt.Errorf("parse license: %w", err)
+	}
+
+	ecDomains := utils.GetDomains(m.releaseData)
+
+	installOpts := kotscli.InstallOptions{
+		AppSlug:      license.Spec.AppSlug,
+		License:      m.license,
+		Namespace:    constants.KotsadmNamespace,
+		ClusterID:    m.clusterID,
+		AirgapBundle: m.airgapBundle,
+		// Skip running the KOTS app preflights in the Admin Console; they run in the manager experience installer when ENABLE_V3 is enabled
+		SkipPreflights:        os.Getenv("ENABLE_V3") == "1",
+		ReplicatedAppEndpoint: netutils.MaybeAddHTTPS(ecDomains.ReplicatedAppDomain),
+	}
+
+	configValuesFile, err := m.createConfigValuesFile(configValues)
+	if err != nil {
+		return fmt.Errorf("creating config values file: %w", err)
+	}
+	installOpts.ConfigValuesFile = configValuesFile
+
+	if m.kotsCLI != nil {
+		return m.kotsCLI.Install(installOpts)
+	}
+
+	return kotscli.Install(installOpts)
+}
+
+// createConfigValuesFile creates a temporary file with the config values
+func (m *appInstallManager) createConfigValuesFile(configValues kotsv1beta1.ConfigValues) (string, error) {
+	// Use Kubernetes-specific YAML serialization to properly handle TypeMeta and ObjectMeta
+	data, err := kyaml.Marshal(configValues)
+	if err != nil {
+		return "", fmt.Errorf("marshaling config values: %w", err)
+	}
+
+	configValuesFile, err := os.CreateTemp("", "config-values*.yaml")
+	if err != nil {
+		return "", fmt.Errorf("create temp file: %w", err)
+	}
+	defer configValuesFile.Close()
+
+	if _, err := configValuesFile.Write(data); err != nil {
+		os.Remove(configValuesFile.Name())
```

Comment:
im a believer in explicitly ignoring the error if that is the intention so it doesnt go unhandled unless it is intentional

```suggestion
		_ = os.Remove(configValuesFile.Name())
```